src/swupd_lib/curl.c

```diff
@@ -457,7 +457,9 @@ enum download_status process_curl_error_codes(int curl_ret, CURL *curl_handle)
 		}
 	} else { /* download failed but let our caller do it */
 		long response = 0;
-		curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response);
+		if (curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response) != CURLE_OK) {
```

Comment:
```suggestion
		if ((curl_ret = curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response)) != CURLE_OK) {
				error("curl_easy_getinfo: %s\n", curl_easy_strerror(curl_ret));
		
```

Reply:
I don't know if we want to be changing curl_ret as this is used in other parts of the function. I agree we might be able to get better error messages though. Let me try another patch.